### ISSUE TITLE
Can't play old random mode game records in full after updating to 2023.1130.0

### ISSUE THREAD

**Issue by LianSheng197** [ppy/osu#25644]
### Type

Crash to desktop

### Bug description

After the update, the random logic of some segments is different from the old version.

### Screenshots or videos

The following video shows a playthrough with Random Mode enabled.

https://github.com/ppy/osu/assets/18535495/27a620e7-775d-4d3e-9103-b3c288a27a2a

The top half of the movie is version `2023.1114.0-lazer`
And the bottom half is `2023.1130.0-lazer`

And the record file: [LianSheng playing GALNERYUS - RAISE MY SWORD (beem2137) [Anzy_s Insane] (2023-11-14_19-37).zip](https://github.com/ppy/osu/files/13533463/LianSheng.playing.GALNERYUS.-.RAISE.MY.SWORD.beem2137.Anzy_s.Insane.2023-11-14_19-37.zip)

### Version

2023.1130.0-lazer

### Logs

[logs_Dec02_2023.zip](https://github.com/ppy/osu/files/13533467/logs_Dec02_2023.zip)

**Comment by peppy** [comment]
Tentatively marking priority 0 as a regression, at least until we investigate how this happened.

**Comment by EVAST9919** [comment]
`OsuModRandom` relies on `NewCombo` information in multiple places and `NewCombo` parsing has been adjusted in https://github.com/ppy/osu/commit/7998204cfe1e529bc684b54f64356882ec2c81a2 (first hitobject after spinner starts a new combo). This is supported by the fact that difference in beatmap are visible after the spinner.

**Comment by bdach** [comment]
Yeah that checks out. I think this is a wontfix then - which is fine given we're wiping leaderboards anyhow.
